h5p-bildetema/src/components/LanguageSelectorElement/LanguageSelectorElement.tsx

```diff
@@ -0,0 +1,106 @@
+import React from "react";
+import { useL10ns } from "use-h5p";
+import { Language, UserData } from "../../../../common/types/types";
+import { defaultFavouriteLanguages } from "../Bildetema/Bildetema";
+import { languages, languagesOriginal } from "../../constants/languages";
+import { AllowedLanguage } from "../../types/AllowedLanguage";
+import styles from "./LanguageSelectorElement.module.scss";
+
+type LanguageSelectorElement = {
+  language: Language;
+  middleElement: boolean;
+  userData: UserData;
+  setUserData: (updatedUserData: UserData) => void;
+  favLanguages: Language[];
+  setFavLanguages: React.Dispatch<React.SetStateAction<Language[]>>;
+  handleChangeLanguage: (newLanguage: Language) => void;
+};
+
+export const LanguageSelectorElement: React.FC<LanguageSelectorElement> = ({
+  language,
+  middleElement,
+  userData,
+  setUserData,
+  favLanguages,
+  setFavLanguages,
+  handleChangeLanguage,
+}) => {
+  const languageKeys = languages.map(
+    lang => `lang_${lang}`,
+  ) as Array<`lang_${AllowedLanguage}`>;
+
+  const translations = useL10ns(...languageKeys, "selectLanguage");
+
+  const [isChecked, setIsChecked] = React.useState(
+    !!favLanguages.find(favLang => favLang.code === language.code),
+  );
+
+  const handleChange = (): void => {
+    const userDataSnapshot = userData;
```

Comment:
When aliasing like this, any update to `userDataSnapshot` will affect `userData`. Was that intentional?
![image](https://user-images.githubusercontent.com/9085189/182082288-ebe68d12-f73c-43f8-9d51-7bd4ffc9c79c.png)

Consider using `structuredClone` to deep copy all fields from `userData` into `userDataSnapshot`.

Reply:
This was done purely to counter the assignment to a function parameter (userData), which was complained about by eslint. 
![image](https://user-images.githubusercontent.com/43214700/182095859-5c88247b-e1d8-483a-897f-19f4cc595eae.png)
Essentially, we're setting the user data in the end of the function, so changing the userData in the process should be fine, unless some weird bugs happen due to the state being manipulated just before a call to setState (setUserData). 
I'll make a copy instead of assigning the pointer to the same object, this should fix any potential problems 👍